### ISSUE TITLE
Nested JOINs with asterisk selector reads data from wrong columns

### ISSUE THREAD

**Issue by ste-art** [yandex/ClickHouse#3509]
ClickHouse server version 18.14.10 revision 54409

Data:
```SQL
CREATE TABLE t (name String, value Int32) ENGINE = Memory()
INSERT INTO t (name, value) VALUES ('a', 1), ('a', 7), ('a', 9)

SELECT * FROM t
┌─name─┬─value─┐
│ a    │     1 │
│ a    │     7 │
│ a    │     9 │
└──────┴───────┘
```
Simple queries:
```SQL
SELECT
    name,
    COUNT() AS less
FROM t
WHERE value < 5
GROUP BY name

┌─name─┬─less─┐
│ a    │    1 │
└──────┴──────┘

SELECT
    name,
    COUNT() AS more
FROM t
WHERE value >= 5
GROUP BY name

┌─name─┬─more─┐
│ a    │    2 │
└──────┴──────┘

SELECT
    name,
    COUNT() AS total
FROM t
GROUP BY name

┌─name─┬─total─┐
│ a    │     3 │
└──────┴───────┘

```
'less' = 1, 'more' = 2, 'total' = 3. Correct.

Simple JOIN:
```SQL
SELECT *
FROM
(
    SELECT
        name,
        COUNT() AS less
    FROM t
    WHERE value < 5
    GROUP BY name
)
ANY INNER JOIN
(
    SELECT
        name,
        COUNT() AS more
    FROM t
    WHERE value >= 5
    GROUP BY name
) USING (name)

┌─name─┬─less─┬─name─┬─more─┐
│ a    │    1 │ a    │    2 │
└──────┴──────┴──────┴──────┘
```
In some reason there are two columns 'name', but the values are correct.

Now nested JOIN:
```SQL
SELECT *
FROM
(
    SELECT
        name,
        COUNT() AS total
    FROM t
    GROUP BY name
)
ANY INNER JOIN
(
    SELECT *
    FROM
    (
        SELECT
            name,
            COUNT() AS less
        FROM t
        WHERE value < 5
        GROUP BY name
    )
    ANY INNER JOIN
    (
        SELECT
            name,
            COUNT() AS more
        FROM t
        WHERE value >= 5
        GROUP BY name
    ) USING (name)
) USING (name)

┌─name─┬─total─┬─name─┬─less─┬─name─┬─more─┐
│ a    │     3 │ a    │    2 │ a    │    1 │
└──────┴───────┴──────┴──────┴──────┴──────┘
```

**'less' = 2 and 'more' = 1. That's not right.**

As a workaround you have to explicitly specify column names instead of asterisk in both nested and outer queries:
```SQL
SELECT
    name,
    total,
    less,
    more
FROM
(
    SELECT
        name,
        COUNT() AS total
    FROM t
    GROUP BY name
)
ANY INNER JOIN
(
    SELECT
        name,
        less,
        more
    FROM
    (
        SELECT
            name,
            COUNT() AS less
        FROM t
        WHERE value < 5
        GROUP BY name
    )
    ANY INNER JOIN
    (
        SELECT
            name,
            COUNT() AS more
        FROM t
        WHERE value >= 5
        GROUP BY name
    ) USING (name)
) USING (name)

┌─name─┬─total─┬─less─┬─more─┐
│ a    │     3 │    1 │    2 │
└──────┴───────┴──────┴──────┘
```
With explicit selectors the values are correct, 
But with asterisks the result is wrong.

**Comment by ztlpn** [comment]
Fixed by #3505

**Comment by ztlpn** [comment]
fix is in v18.14.12
